### PR TITLE
feat: added new contract to showcase assembly address operations (#459)

### LINKED ISSUE #459: [Solidity Support] Assembly Address operations
### Problem

The Hedera network supports solidity as one of the smart contracts development languages.
Occasionally items have shown up that surfaced in-equivalence at the language layer.

### Solution

Add examples for assembly address operations

- Add `/contracts/solidity/address/AssemblyAddress.sol`
- Cover the tasklist below

```[tasklist]
### Assembly address Tasks
- [x] `extcodesize(address)`
- [x] `extcodecopy(address)`
- [x] `extcodehash(address)`
```

 Ref: https://docs.soliditylang.org/en/latest/units-and-global-variables.html & https://docs.soliditylang.org/en/latest/assembly.html

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by quiet-node**
**Description**:

- Added a new Solidity contract to demonstrate assembly address operations, including `extcodesize`, `extcodehash`, and `extcodecopy`.
- Included unit tests to verify the functionality of the contract.

**Related issue(s)**: #459

Fixes #459

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: contracts/solidity/address/AssemblyAddress.sol [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+            code := mload(0x40)
+            // new "memory end" including padding
+            mstore(0x40, add(code, and(add(add(size, 0x20), 0x1f), not(0x1f))))
+            // store length in memory
+            mstore(code, size)
+            // actually retrieve the code, this needs assembly
+            extcodecopy(addr, add(code, 0x20), 0, size)
+        }
+    }
+}
```
Comment (ebadiere) [inline contracts/solidity/address/AssemblyAddress.sol:32]: Nit: Blank line at the end?

File: test/solidity/address/AssemblyAddress.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+      assemblyAddressContract.address
+    )
+  })
+
+  it("Should get contract's code size at contract address", async () => {
+    const contractCodeSize = await assemblyAddressContract.codesizeat(
+      assemblyAddressContract.address
+    )
+
+    const expectedContractCodeSize =
```
Comment (Nana-EC) [inline test/solidity/address/AssemblyAddress.js:44]: nit: Good place to add a comment to explain your logic and arithmetic

**Review by Nana-EC — APPROVED** [review]
LG
One nit

File: test/solidity/address/AssemblyAddress.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+const { expect } = require('chai')
+const { ethers } = require('hardhat')
+const Constants = require('../../constants')
+
+describe('AssemblyAddress tests', () => {
```
Comment (Nana-EC) [inline test/solidity/address/AssemblyAddress.js:25]: Add one of the `@solidityevmequivx` tags so this gets picked up for runs

**Review by Nana-EC — APPROVED** [review]
LG.
Add the tag to this or next PR

**Review by Nana-EC — APPROVED** [review]

**Pull request merged**